### PR TITLE
Login with cookie

### PR DESCRIPTION
**Pull request by dstr951**
added a new endpoint "/login" with available post request.
this post request sends an HTTPOnly cookie with a successful response

### CODE REVIEW

File: Server/API/login/postLogin.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	cookie := new(http.Cookie)
+	cookie.Name = "userId"
+	u := new(entities.LoginRequest)
+	if err = c.Bind(u); err != nil {
+		return err
+	}
+	userName := entities.LoginRequest{
+		UserName: u.UserName,
+	}
+	cookie.Value, _ = createToken(userName)
```
Comment (y-golde) [inline Server/API/login/postLogin.go]: _ is for error?

File: Server/API/login/postLogin.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+func postLogin(c echo.Context) (err error) {
+	cookie := new(http.Cookie)
+	cookie.Name = "userId"
+	u := new(entities.LoginRequest)
+	if err = c.Bind(u); err != nil {
+		return err
+	}
+	userName := entities.LoginRequest{
```
Comment (y-golde) [inline Server/API/login/postLogin.go]: Why is the constant userName made of a login request?

File: Server/API/login/postLogin.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		UserName: u.UserName,
+	}
+	cookie.Value, _ = createToken(userName)
+	cookie.HttpOnly = true
+	c.SetCookie(cookie)
+	return c.String(http.StatusOK, "login successful")
```
Comment (y-golde) [inline Server/API/login/postLogin.go]: No need to send a string back

File: Server/API/login/postLogin.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func createToken(userName entities.LoginRequest) (string, error) {
+	var err error
+	atClaims := jwt.MapClaims{}
+	atClaims["authorized"] = true
+	atClaims["user_name"] = userName.UserName
+	atClaims["exp"] = time.Now().Add(time.Hour * 24).Unix()
+	at := jwt.NewWithClaims(jwt.SigningMethodHS256, atClaims)
+	token, err := at.SignedString([]byte(os.Getenv("ACCESS_SECRET")))
```
Comment (y-golde) [inline Server/API/login/postLogin.go:43]: Send me the env

File: Server/Entities/LoginRequest.go [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package entities
+
+type LoginRequest struct {
+	UserName string `json:"userName" bson:"userName"`
```
Comment (y-golde) [inline Server/Entities/LoginRequest.go]: Change bson and saving on client to user_name

File: Server/API/login/postLogin.go (hunk 1 above)
Comment (dstr951) [reply Server/API/login/postLogin.go]: woops I wanted to reference that error later and I forgot
I will add it now

File: Server/Entities/LoginRequest.go (hunk 5 above)
Comment (dstr951) [reply Server/Entities/LoginRequest.go]: so the tag should be `json:"userName" bson:"user_name"`?

File: Server/API/login/postLogin.go (hunk 4 above)
Comment (dstr951) [reply Server/API/login/postLogin.go:43]: Add "ACCESS_SECRET=jdnfksdmfksd" to the server env

File: Server/API/login/postLogin.go (hunk 3 above)
Comment (dstr951) [reply Server/API/login/postLogin.go]: changed it in newer commit

File: Server/API/login/postLogin.go (hunk 2 above)
Comment (dstr951) [reply Server/API/login/postLogin.go]: Good point, changed it in a newer commit

File: Server/API/login/postLogin.go (hunk 1 above)
Comment (dstr951) [reply Server/API/login/postLogin.go]: changed it in a newer commit

File: Server/Entities/LoginRequest.go (hunk 5 above)
Comment (y-golde) [reply Server/Entities/LoginRequest.go]: yea, and save it in the server as user_name

Comment (dstr951) [reply Server/Entities/LoginRequest.go]: when I'll get to saving it to the server :)

Comment (dstr951) [reply Server/Entities/LoginRequest.go]: fixed

File: Server/API/login/postLogin.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+func postLogin(c echo.Context) (err error) {
+	cookie := new(http.Cookie)
+	cookie.Name = "userId"
+	u := new(entities.LoginRequest)
+	if err = c.Bind(u); err != nil {
+		return err
+	}
+	id := uuid.New()
```
Comment (dstr951) [inline Server/API/login/postLogin.go:21]: @y-golde review this new part please, I started using uuid package

**Review by y-golde — APPROVED** [review]

File: Server/API/login/postLogin.go (hunk 6 above)
Comment (y-golde) [reply Server/API/login/postLogin.go:21]: approved

**Pull request merged**